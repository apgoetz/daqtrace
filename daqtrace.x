SECTIONS
{
  /* `0` specifies the start address of this virtual (`(INFO)`) section */
  .daqtrace 0 (INFO) :
  {

    /* Everything user-defined */
    *(.daqtrace);
  }
}

ASSERT(SIZEOF(.daqtrace) < (1 << 16), ".daqtrace section id must fit in u16");
