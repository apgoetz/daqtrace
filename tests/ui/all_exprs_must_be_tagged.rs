use daqtrace::trace;
mod common;


// all trace args must be tagged
fn main() {
    let sink = common::DummySink;
    trace!(sink,hello,123);
}

