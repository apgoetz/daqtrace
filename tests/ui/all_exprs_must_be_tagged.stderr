error: trace! macro requires that all logged expressions be tagged with a label for example, trace!(_,_, my_label_x : 1 + 1); or be a simple expression, for example, trace!(_,_,myvarname)
 --> $DIR/all_exprs_must_be_tagged.rs:8:23
  |
8 |     trace!(sink,hello,123);
  |                       ^^^
